compiler/x/codegen/OutlinedInstructions.cpp

```diff
@@ -175,18 +120,34 @@ void TR_OutlinedInstructions::generateOutlinedInstructionsDispatch()
    if (_targetReg)
       {
       TR_ASSERT(resultReg, "assertion failure");
+      TR_ASSERT(resultReg->getKind() == _targetReg->getKind(), "OutlinedInstructions: targetReg must have same time as resultReg.");
+
       TR::RegisterPair *targetRegPair = _targetReg->getRegisterPair();
       TR::RegisterPair *resultRegPair =  resultReg->getRegisterPair();
+
+      TR_ASSERT((targetRegPair == NULL && resultRegPair == NULL) ||
+                (targetRegPair != NULL && resultRegPair != NULL),
+                "OutlinedInstructions: targetReg and resultReg must be both register pairs or neither register pairs.");
       if (targetRegPair)
          {
-         TR_ASSERT(resultRegPair, "OutlinedInstructions: targetReg is a register pair and resultReg is not");
-         generateRegRegInstruction(_targetRegMovOpcode, _callNode, targetRegPair->getLowOrder(),  resultRegPair->getLowOrder(),  _cg);
-         generateRegRegInstruction(_targetRegMovOpcode, _callNode, targetRegPair->getHighOrder(), resultRegPair->getHighOrder(), _cg);
+         generateRegRegInstruction(MOVRegReg(), _callNode, targetRegPair->getLowOrder(),  resultRegPair->getLowOrder(),  _cg);
+         generateRegRegInstruction(MOVRegReg(), _callNode, targetRegPair->getHighOrder(), resultRegPair->getHighOrder(), _cg);
          }
       else
          {
-         TR_ASSERT(!resultRegPair, "OutlinedInstructions: resultReg is a register pair and targetReg is not");
-         generateRegRegInstruction(_targetRegMovOpcode, _callNode, _targetReg, resultReg, _cg);
+         switch (resultReg->getKind())
+            {
+            case TR_GPR:
+               generateRegRegInstruction(MOVRegReg(), _callNode, _targetReg, resultReg, _cg);
+               break;
+            case TR_FPR:
+            case TR_VRF:
+               generateRegRegInstruction(MOVDQURegReg, _callNode, _targetReg, resultReg, _cg);
+               break;
+            default:
+               TR_ASSERT(false, "OutlinedInstructions: unsupported result register kind.");
+               break;
+            }
```

Comment:
You'll save some code space by having the `switch` just select the opcode and then having one `generateRegRegInstruction` call after the switch.

Reply:
Sure, updated.